syft/source/directory_resolver.go

```diff
@@ -276,15 +306,22 @@ func (r *directoryResolver) AllLocations() <-chan Location {
 func (r *directoryResolver) FileMetadataByLocation(location Location) (FileMetadata, error) {
 	info, exists := r.infos[location.ref.ID()]
 	if !exists {
-		return FileMetadata{}, fmt.Errorf("location: %+v : %w", location, os.ErrExist)
+		return FileMetadata{}, fmt.Errorf("location: %+v : %w", location, os.ErrNotExist)
+	}
+
+	uid := -1
+	gid := -1
+	if stat, ok := info.Sys().(*syscall.Stat_t); ok {
```

Comment:
good call 👍

Reply:
@wagoodman  I totally agree, sharing a _directoryResolver_ is probably simpler than sharing a _FileTree_ pointer.
Originally thought about sharing the _directoryResolver_ pointer but i was not sure how it will effect parts i have not yet explored.

Can you please clarify a bit on the _Mutex_ comment? 
It seems to me that even if _source.FileResolver()_ creates only one directoryResolver (Stores pointer in struct) the initialization process still can create a race (Two threads initialization the first directoryResolver)
Eg:
```
// Called by catalogers child threads concurrently 
func (s Source) FileResolver(scope Scope) (FileResolver, error) {
	switch s.Metadata.Scheme {
	case DirectoryScheme:
                if s.dirResolver == nil {
                          // Not thread safe
		          resolver, err := newDirectoryResolver(s.Tree, s.Metadata.Path)
                          if err == nil {
                                 s.dirResolver = resolver
                          }
                }
                return s.dirResolver, err
...
...
```
* Maybe we can initialize the directory resolver in the source _New()_ constructor function? (Called by main thread)?
* Maybe leave the Mutex and only replace the FileTree pointer with a directoryResolver pointer?